Use Array.Empty in WebEncoders

martincostello opened the pull request:
Use `Array.Empty<byte>()` instead of allocating a new empty byte array.

martincostello committed:
Use Array.Empty<T>()

Use Array.Empty<byte>() instead of allocating a new empty array.

martincostello committed:
Use Array.Empty<T>()

Use Array.Empty<ExceptionDetails>() instead of allocating a new empty array.

Tratcher commented on src/Shared/WebEncoders/WebEncoders.cs:
```diff
@@ -26,7 +26,7 @@ namespace Microsoft.Extensions.Internal
     static class WebEncoders
     {
-        private static readonly byte[] EmptyBytes = new byte[0];
+        private static readonly byte[] EmptyBytes = Array.Empty<byte>();
```
Do we even need this field anymore? Can Array.Empty replace it everywhere?

martincostello replied:
I did consider that, but wasn’t sure if there was a reason it was using the field rather than calling the method at runtime so left it. Happy to remove it and call Empty() at the call sites instead.

Tratcher replied:
The old one was caching the allocation, but I think Array.Empty does that internally.

martincostello committed:
Remove EmptyBytes field

Remove the field containing an empty byte[] and just call Array.Empty<byte>() instead.

Tratcher commented:
Thanks
